acvm/src/compiler/transformers/csat.rs

```diff
@@ -188,6 +188,42 @@ impl CSatTransformer {
         new_gate
     }
 
+    /// Normalise an expression by dividing it with its first coefficient
+    /// The first coefficient here means linear_combinations[0].0 of mul_terms[0].0 if the linear_combinations is empty
+    /// The function panic if the input expression is constant
+    fn normalise(expr: &Expression) -> (FieldElement, Expression) {
+        let mut norm = expr.clone();
```

Comment:
```suggestion
    fn normalise(expr: Expression) -> (FieldElement, Expression) {
```

We can pass down ownership of the intermediate gate rather than always cloning it as this is the only function which needs ownership of it.

Reply:
Done